controller/ResultController.php

```diff
@@ -0,0 +1,169 @@
+                        <imsx_description>{{description}}</imsx_description>
+                        <imsx_messageRefIdentifier>{{messageIdentifier}}</imsx_messageRefIdentifier>
+                        <imsx_operationRefIdentifier>replaceResult</imsx_operationRefIdentifier>
+                    </imsx_statusInfo>
+                </imsx_POXResponseHeaderInfo>
+            </imsx_POXHeader>
+            <imsx_POXBody>
+                <replaceResultResponse />
+            </imsx_POXBody>
+        </imsx_POXEnvelopeResponse>
+    ';
+
+    private $dom;
+
+    public function actionResultScore()
+    {
+        $payload = $this->getRequestParameter('payload');
+        $this->manageResult($payload);
+    }
+
+    /**
+     * Create a list or a list element
+     * @return Response
+     * @throws common_exception_BadRequest
+     */
+    public function manageResult($payload)
+    {
+        if (!$this->isXmlHttpRequest()) {
+            // throw new common_exception_BadRequest('wrong request mode');
+        }
+
+        $scoreDescriptionTemplate = 'Score for {sourceId} is now {score}';
+
+        $statuses = array(
+            400 => 'Invalid score',
+            404 => 'DeliveryExecution not found',
+            501 => 'Method not implemented (for other requests than replaceResultRequest)',
+            500 => 'Other internal error => client should retry',
+        );
+
+        try {
+            $this->dom = new \DOMDocument();
+            $this->dom->loadXML($payload);
+            $this->validateResultRequest($payload);
+        } catch (\Exception $e) {
+//            $this->logError('Request XML does not have replaceResultRequest element');
+            return $this->sendResponse([
+                '{{codeMajor}}' => self::FAILURE_MESSAGE,
+                '{{description}}' =>$statuses[501],
+                '{{messageId}}' => '',
+            ], 501);
+        }
+
+        $result = $this->getResult();
+
+        if (!$this->isScoreValid($result['score'])) {
+//            $this->logError('Score is not in the range [0..1]');
+            return $this->sendResponse([
+                '{{codeMajor}}' => self::FAILURE_MESSAGE,
+                '{{description}}' => $statuses[400],
+                '{{messageId}}' => $result['messageIdentifier'],
+            ], 400);
+        }
+
+        try {
+            $resultService = $this->getServiceManager()->get(ResultService::SERVICE_ID);
+            $deliveryExecution = $resultService->getDeliveryExecutionById($result['sourcedId']);
+        } catch (\Exception $e) {
+//            $this->logError('Delivery Execution with ID ' . $sourcedId);
+            return $this->sendResponse([
+                '{{codeMajor}}' => self::FAILURE_MESSAGE,
+                '{{description}}' => $statuses[404],
+                '{{messageId}}' => $result['messageIdentifier'],
+            ], 404);
+        }
+    }
+
+    private function validateResultRequest()
```

Comment:
Add $dom as a function parameter. You already give it on the function call line 89.